tls_codec/derive/tests/decode.rs

```diff
@@ -549,6 +549,13 @@ mod conditional_deserialization {
         assert_eq!(deserializable_struct.a, undeserializable_struct.a);
         assert_eq!(deserializable_struct.b, undeserializable_struct.b);
 
+        #[conditionally_deserializable]
+        #[derive(TlsSize, TlsSerialize, PartialEq, Debug)]
+        struct NestedExampleStruct {
```

Comment:
This seems to fail when adding a generic member. Is this supposed to work?
It's ok when not, but should be documented then and throw a useful error.